### PR TITLE
RAMClustR experimental design input

### LINKED ISSUE #216: Add tool which resembles the `defineExperiment(...)` functionality
A wrapper which allows to fill in Experimental design. See [here](https://github.com/RECETOX/RAMClustR/issues/2) for more info.

### PR DESCRIPTION
**Pull request by xtrojak**
Added tool to specify `RAMClustR` experimental design. This produces a `.csv` file, which can be provided to `RAMClustR` as additional annotation information. Such file is processed by [`defineExperiment`](https://github.com/RECETOX/RAMClustR/blob/master/R/defineExperiment.R) function.

Extended `RAMClustR` itself by creating new input field for such a file. This file is then read by mentioned R function and passed to `RAMClustR` as an Robject.

Close #216, which covers one of the last tasks from #35.

### CODE REVIEW

**Review by hechth — APPROVED** [review]

File: tools/ramclustr/ramclustr_define_experiment.xml [hunk 1]
```diff
@@ -0,0 +1,198 @@
+<tool id="ramclustr_define_experiment" name="RAMClustR define experiment" version="1.0.0+galaxy0">
```
Comment (martenson) [inline tools/ramclustr/ramclustr_define_experiment.xml:1]:
```suggestion
<tool id="ramclustr_define_experiment" name="RAMClustR define experiment" version="1.0.0">
```

**Comment by martenson** [comment]
Looks very good, well done.

File: tools/ramclustr/ramclustr_define_experiment.xml (hunk 1 above)
Comment (hechth) [reply tools/ramclustr/ramclustr_define_experiment.xml:1]: Since this is a "galaxy only" tool, we drop the two-part versioning? Makes sense!

Comment (martenson) [reply tools/ramclustr/ramclustr_define_experiment.xml:1]: Indeed, there is no other dependency that we need to harmonize with, this change makes it clear.

File: tools/ramclustr/ramclustr_define_experiment_wrapper.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+
+def main(argv):
+    parser = argparse.ArgumentParser(description="RAMClustR define experiment.")
+
+    for arg in ARGUMENTS.keys():
+        parser.add_argument(f"--{arg}", type=ARGUMENTS[arg]["type"], help=ARGUMENTS[arg]["help"])
+
+    parser.add_argument("--output_file", type=str, help="Path to output csv file.")
+    args = parser.parse_args().__dict__
```
[the remark this answers is not in the document]
Comment (xtrojak) [reply tools/ramclustr/ramclustr_define_experiment_wrapper.py]: Good point! Fixed in f5c44b5.

File: tools/ramclustr/ramclustr_define_experiment.xml (hunk 1 above)
Comment (xtrojak) [reply tools/ramclustr/ramclustr_define_experiment.xml]: Yes, makes sense, thank you!

**Pull request merged**